cmd/clusterctl/client/repository/overrides.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/util/homedir"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+)
+
+const (
+	overrideFolder    = "overrides"
+	overrideFolderKey = "overridesFolder"
+)
+
+// Overrider provides behavior to determine the overrides layer.
+type Overrider interface {
+	Path() string
+}
+
+type overrides struct {
+	configVariablesClient config.VariablesClient
+	providerLabel         string
+	version               string
+	filePath              string
+}
+
+type overrideInfo struct {
```

Comment:
rename into NewOverrideOptions (or NewOverrideInput) for consistency with the other structs used as input for methods?